go-controller/pkg/node/gateway_init.go

```diff
@@ -74,7 +74,7 @@ func bridgedGatewayNodeSetup(nodeName, bridgeName, bridgeInterface, physicalNetw
 }
 
 // getNetworkInterfaceIPAddresses returns the IP addresses for the network interface 'iface'.
-func getNetworkInterfaceIPAddresses(iface string) ([]*net.IPNet, error) {
+func getNetworkInterfaceIPAddresses(iface string, kubeNodeIP *net.IP) ([]*net.IPNet, error) {
```

Comment:
@andreaskaris if we're just going to de-reference kubeNodeIP later, maybe we should just make it a plain net.IP. I think here that's fine because it's not a big struct, and this isn't called that much.